react/Labs/PasswordInput/helpers.js

```diff
@@ -0,0 +1,56 @@
+// Return given password srength as an object {percentage, label}
+export const getStrength = password => {
+  if (!password && password !== '') {
+    throw new Error('password parameter is missing')
+  }
+  if (!password.length) {
+    return { percentage: 0, label: 'weak' }
+  }
+
+  const charsets = [
+    // upper
+    { regexp: /[A-Z]/g, size: 26 },
+    // lower
+    { regexp: /[a-z]/g, size: 26 },
+    // digit
+    { regexp: /[0-9]/g, size: 10 },
+    // special
+    { regexp: /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/g, size: 30 }
+  ]
+
+  const possibleChars = charsets.reduce(function(possibleChars, charset) {
+    if (charset.regexp.test(password)) possibleChars += charset.size
+    return possibleChars
+  }, 0)
+
+  const passwordStrength =
+    Math.log(Math.pow(possibleChars, password.length)) / Math.log(2)
+
+  // levels
+  const _at33percent = 50
+  const _at66percent = 100
+  const _at100percent = 150
```

Comment:
I think I recognize this from zxcvbn, is it the same extracted from there? If so did you consider straight up using zxcvbn as a dependency?

Reply:
Actually I extracted it from [cozy-settings](https://github.com/cozy/cozy-settings/blob/master/src/lib/passwordHelper.js). I don't know if that's code extracted from zxcvbn, so I never considered using zxcvbn as a dependency.